Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp

```diff
@@ -356,61 +358,42 @@ static void HotplugThreadFunc()
       }
     }
 
-    sf::SocketSelector selector;
-    for (auto& server : s_servers)
-    {
-      selector.add(server.m_socket);
-    }
-
-    using namespace std::chrono;
-    using namespace std::chrono_literals;
-    const auto timeout = s_next_listports - SteadyClock::now();
-
-    // Selector's wait treats a timeout of zero as infinite timeout, which we don't want
-    const auto timeout_ms = std::max(duration_cast<milliseconds>(timeout), 1ms);
-    if (!selector.wait(sf::milliseconds(timeout_ms.count())))
-    {
-      continue;
-    }
-
-    for (auto& server : s_servers)
-    {
-      if (!selector.isReady(server.m_socket))
-      {
-        continue;
-      }
-
-    // Equally spread the timeout between the number of servers.
-    // As of now SERVER_LISTPORTS_INTERVAL is 1 second, so that should be plenty for a lot of
-    // servers. If we don't do this, the first server that is unresponsive will take away from the
-    // time of the following ones.
+    // Equally spread the timeout between the number of servers to avoid waits being too long
+    // and hanging the main thread on closure.
     // We use the "now" from above to avoid breaking while stepping through the code.
     // Note that this thread won't be running if we have no servers.
     const auto timeout = (s_next_listports_time - now) / s_servers.size();
-    // ReceiveWithTimeout treats a timeout of zero as infinite timeout, which we don't want.
-    // Also define a min/max wait time for every server, to make sure this works if we have many,
-    // and that we don't hang the main thread too long when closing this one.
-    const auto adjusted_timeout_ms =
-        std::clamp(duration_cast<milliseconds>(timeout), SERVER_MIN_LISTPORTS_INTERVAL,
-                   SERVER_MAX_LISTPORTS_INTERVAL);
 
     // Receive controller port info within a time from our request.
     // Run this even if we sent no new requests, to disconnect devices,
     // sleep (wait) the thread and catch old responses.
     for (auto& server : s_servers)
     {
-      auto current_timeout_ms = adjusted_timeout_ms;
-      if (auto port_info = msg.CheckAndCastTo<Proto::MessageType::PortInfo>())
+      // ReceiveWithTimeout treats a timeout of zero as infinite timeout, which we don't want.
+      // Also define a max wait time for every server, again, to avoid long waits.
+      const auto adujusted_timout_ms =
```

Comment:
Still wrapping my  head around why this will be helpful over a 1ms wait that is shared by all sockets but this logic isn't dependent on the server, it can be moved out of the loop

Reply:
You are reviewing an old version, this isn't there anymore.